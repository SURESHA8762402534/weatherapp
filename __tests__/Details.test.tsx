import { fireEvent, render, screen } from "@testing-library/react-native";
import Deatails from "../src/Details";

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

jest.setTimeout(15000);
test("test for display Details page", async () => {
  const tree = render(<Deatails />);
  expect(tree).toMatchSnapshot();

  await new Promise((r) => setTimeout(r, 10000));
  // const test1 =
  const test = screen.findByTestId("test-0");
  expect(test).toBeTruthy();
  const btn2 = await screen.getByTestId("capital");
  fireEvent.press(btn2);
  const btn3 = screen.getByTestId("back1");
  fireEvent.press(btn3);
  const btn1 = screen.getByTestId("back");
  fireEvent.press(btn1);
});
