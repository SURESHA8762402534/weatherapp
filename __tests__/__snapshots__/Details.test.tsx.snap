// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for display Details page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
        "marginTop": "40px",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 10,
          "margin": "10px",
          "width": "30%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="back"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "lightblue",
                },
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View>
          <Text>
            Somthing went Wrong
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  />
</RCTSafeAreaView>
`;

exports[`test for display content page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
        "marginTop": "40px",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 10,
          "margin": "10px",
          "width": "30%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="back"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "lightblue",
                },
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View>
          <Text>
            Somthing went Wrong
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  />
</RCTSafeAreaView>
`;
